code/datums/mind.dm

```diff
@@ -88,8 +88,22 @@
 				qdel(i)
 		antag_datums = null
 	QDEL_NULL(language_holder)
+	set_current(null)
 	return ..()
 
+/datum/mind/proc/set_current(mob/new_current)
+	if(new_current && QDELETED(new_current))
```

Comment:
QDELETED checks for null values 
```suggestion
	if(QDELETED(new_current))
```

Reply:
I'm not sure what the deal with that change is, but

```
#define QDELETED(X) (!X || QDELING(X))
```

That ends up evaluating to `X && (!X || QDELING(X))`, which means the if **ensures** there is a `new_current` rather than checks in case one is missing.

Reply:
Ah, this might be so that you can specifically assign `null` to remove any mobs from a mind

Reply:
Can't you just do QDELING(X) then?

Reply:
That'd still require the code to be `new_current && QDELING(new_current)`, but that should work and remove a bit of redundancy. I don't think I've seen `QDELING` used directly in the codebase, but if you think it's better I can swap it out.

Reply:
Changed to `QDELING`